syntax = "proto3";

package auth;

service AuthService {
  rpc Login(LoginRequest) returns (LoginResponse);
  rpc Register(RegisterRequest) returns (RegisterResponse);
  rpc ValidateToken(ValidateTokenRequest) returns (ValidateTokenResponse);
}

message RegisterRequest {
  string username = 1;
  string password = 2;
}

message ValidateTokenRequest {
  string token = 1;
}

message LoginRequest {
  string username = 1;
  string password = 2;
}

message RegisterResponse {
  bool success = 1;
  string token = 2;
  string message = 3; // Optional message for success or failure
}

message LoginResponse {
  bool success = 1;
  string token = 2;
}

message ValidateTokenResponse {
  bool valid = 1;
}